hexrd/ui/image_load_manager.py

```diff
@@ -40,70 +41,77 @@ def __init__(self):
         super(ImageLoadManager, self).__init__(None)
         self.unaggregated_images = None
 
-    def check_images(self, fnames):
-        dets = HexrdConfig().detector_names
-        files = [[] for i in range(len(dets))]
-        core_name = os.path.split(fnames[0])[1]
-        for det in dets:
-            if det in fnames[0]:
-                core_name = core_name.replace(det, '')
-        matches = 0
-        for file in fnames:
-            file_name = os.path.split(file)[1]
-            for det in dets:
-                if (det in file_name) and (core_name == file_name.replace(det, '')):
-                    matches += 1
-                    pos = dets.index(det)
-                    files[pos].append(file)
-        # Display error if selected files do not match ea. detector
-        if not len(fnames) == matches:
-            msg = ('ERROR - Files must contain detector name.')
-            QMessageBox.warning(None, 'HEXRD', msg)
-            return []
-        else:
-            return files
+    def load_images(self, fnames):
+        files = self.explict_selection(fnames)
+        manual = False
+        if not files:
+            files = self.match_files(fnames)
+            matched = self.check_success(files)
+            if not matched:
+                manual = True
+                files = [[fname] for fname in fnames]
+        return files, manual
+
+    def check_success(self, files):
+        detectors = HexrdConfig().detector_names
+        # Make sure there are the same number of files for each detector
+        # and at least one file per detector
+        if (not files[0]
+                or len(files) != len(detectors)
+                or any(len(files[0]) != len(elem) for elem in files)):
+            return False
+        # If the files do not contain detector names they will need to
+        # be manually matched
+        if not all(any(d in f for d in detectors) for f in files[0]):
+            return False
+        return files
```

Comment:
It seems that this function either returns the `files` or `False`? May be it should just return `True` or `False`?

Reply:
Fixed